### PR TITLE
Add script to build LLVM + --with-cmodel=medany swtich

### PR DESCRIPTION
**Pull request by StanSofomoDev**
2 changes here:

1. I've made building your own toolchain for baremetal mandatory, as you need to use `--with-cmodel=medany` for it to work with benchmarks.

2. I've wrapped this process into a script, that just builds LLVM for both Baremetal and Linux.

### CODE REVIEW

File: LLVM_FOR_RISCV.md [hunk 1]
```diff
@@ -158,15 +181,3 @@ Linux:
 ```bash
 [PATH_TO_YOUR_LLVM_LINUX_INSTALL]/bin/clang --target=riscv64-unknown-linux-gnu -o hello hello.c
```
Comment (jeffnye-gh) [inline LLVM_FOR_RISCV.md:182]: isnt this path known due to completing the install?

**Review by jeffnye-gh — CHANGES_REQUESTED** [review]
I have comments, my build is running, I expect it to finish without error.

File: LLVM_FOR_RISCV.md (hunk 1 above)
Comment (StanSofomoDev) [reply LLVM_FOR_RISCV.md:182]: it is, but it's just an example, after changes to script this will be given by user

File: scripts/build_llvm.sh [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+if [ -z "$TOP" ] || [ ! -d "$TOP" ]; then
+    echo "The TOP variable is not set to a valid directory."
+    echo "To set the required environment variables, cd into your work area and run: source how-to/env/setuprc.sh"
+    exit 1
+fi
+
+read -p "Enter the install path [default is $TOP]: " INSTALL_PATH
+INSTALL_PATH=${INSTALL_PATH:-$TOP}
+
```
Comment (jeffnye-gh) [inline scripts/build_llvm.sh:16]: jeffnye@compute2:/data/users/jeffnye/condor$ bash how-to/scripts/build_llvm.sh
Enter the install path [default is /data/users/jeffnye/condor]: 

That is not where it gets installed, it is actually  /data/users/jeffnye/condor/llvm

This is not hard, get this right and so this task can be closed.

File: scripts/build_llvm.sh [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    mkdir -p "$INSTALL_PATH/llvm/llvm-linux" || { echo "Failed to create llvm-linux directory. Exiting."; exit 1; }
+    echo "Using install path: $INSTALL_PATH"
+fi
+
+read -p "Enter the source directory path [default is $TOP]: " SOURCE_DIR
+SOURCE_DIR=${SOURCE_DIR:-$TOP}
+
+if [ -d "$SOURCE_DIR/llvm_source" ]; then
+    echo "Source directory already exists. Exiting."
+    exit 1
```
Comment (jeffnye-gh) [inline scripts/build_llvm.sh:32]: more of the same.

**Review by jeffnye-gh — CHANGES_REQUESTED** [review]
see the comments.

File: LLVM_FOR_RISCV.md [hunk 4]
```diff
@@ -40,100 +40,16 @@ successfully deactivated the environments.
...
-
-## The RISC-V GNU Toolchain for Baremetal
-
-The `RISC-V GNU toolchain` is necessary to use the LLVM, you should be able to find it under `/data/tools/riscv64-unknown-elf`. Copy it to the `_install` directory in your workspace
+**Running the Script:**
 
 ```bash
-cp -fa /data/tools/riscv64-unknown-elf/* _install/
+cd $TOP
+bash how-to/scripts/build_llvm.sh
```
Comment (jeffnye-gh) [inline LLVM_FOR_RISCV.md:49]: These are the instructions, 'here run this script'?

this is a giant process, maybe a bit more explanation is in order.

**Review by jeffnye-gh — CHANGES_REQUESTED** [review]
read me needs some explanation to a user who is going to wait an hour? or more

**Review by jeffnye-gh — APPROVED** [review]
works great thanks for the hard work

please try to merge this, you should be able on your own, let's test it. 

i will close the ticket.

**Pull request merged**